### PR TITLE
Fixes #1447

### LINKED ISSUE #1447: Add script for removing whitespaces
A script can be added to remove trailing white spaces from ```python``` files in the repository.

### PR DESCRIPTION
**Pull request by anikamurarka**
Automatically formats all the ```python``` files to ```pep8``` format.

### CODE REVIEW

**Comment by The-Compiler** [comment]
This seems to be the equivalent of simply running `autopep8 -ir .` which does a whole lot more than removing trailing whitespaces, with a lot of stuff I don't quite agree with :wink:

To be honest I disagree with the goal of `autopep8` in general, as sometimes a human judgement simply is better than a tool (which is why I prefer to *check* things using pep8/pylint instead).

Would you still be interested in adding a script (to `dev/scripts`) which *only* removes trailing whitespaces?

File: format_py.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                source_code = afile.read()
+                afile.seek(0)
+                afile.truncate()
+                afile.write(autopep8.fix_code(source_code))
+                afile.close()
+
+
+if __name__ == "__main__":
+    path_of_dir = [
+        os.path.dirname(BASE_DIR),
```
Comment (The-Compiler) [inline format_py.py]: This also runs `autopep8` over directories like `.venv` where it fails:

```
Traceback (most recent call last):
  File "ap8.py", line 28, in <module>
    fix_file(x)
  File "ap8.py", line 15, in fix_file
    afile = open(filename, 'r+')
PermissionError: [Errno 13] Permission denied: '/home/florian/proj/qutebrowser/git/.venv/lib/python3.4/weakref.py'
```

File: format_py.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+BASE_DIR = os.path.abspath(__file__)
+
+
+def fix_file(path):
+    for dirpath, _, filenames in os.walk(path):
+        for f in filenames:
+            if f.endswith('.py'):
+                filename = os.path.abspath(os.path.join(dirpath, f))
+                afile = open(filename, 'rw+')
```
Comment (The-Compiler) [inline format_py.py]: This won't work:

```
Traceback (most recent call last):
  File "ap8.py", line 28, in <module>
    fix_file(x)
  File "ap8.py", line 15, in fix_file
    afile = open(filename, 'rw+')
ValueError: must have exactly one of create/read/write/append mode
```

**Comment by anikamurarka** [comment]
Okay, @The-Compiler , will make the required changes.

**Comment by The-Compiler** [comment]
As already mentioned in https://github.com/The-Compiler/qutebrowser/pull/1452#issuecomment-214410329 I'm not comfortable with any solution using `autopep8` as it does a whole lot than what's mentioned in #1447 (removing line endings).

I'd be fine with a small script only removing trailing whitespace if you think it's needed, @rumpelsepp showed an example in https://github.com/The-Compiler/qutebrowser/issues/1447#issuecomment-214635800.

I'm going to close this to avoid any more misunderstandings - please don't take that personally though!

**Comment by anikamurarka** [comment]
Sorry @The-Compiler, I missed on that comment :(, I just went through the code errors. I will work on a descent script for trailing spaces only. 
Sorry again for messing up here :sweat_smile:

**Comment by rumpelsepp** [comment]
Just use sth. linke this:

``` sh
#!/bin/sh

find . -type f -name '*.py' -exec sed --in-place 's/[[:space:]]\+$//' {} \+
```

http://stackoverflow.com/a/10711226/2587286

Or maybe also useful, I have this in my gitconfig:

```
[alias]
    # http://stackoverflow.com/a/15398512/2587286
    # The 'git stash save' fails if the tree is clean (instead of
    # creating an empty stash :P). So, we only 'stash' and 'pop' if
    # the tree is dirty.
    #
    # The 'git rebase --whitespace=fix HEAD~' throws away the commit
    # if it's empty, and adding '--keep-empty' prevents the whitespace
    # from being fixed. So, we first check that the index is dirty.
    #
    # Also:
    # - '(! git diff-index --quiet --cached HEAD)' is true (zero) if
    #   the index is dirty
    # - '(! git diff-files --quiet .)' is true if the tree is dirty
    #
    # The 'rebase --whitespace=fix' trick is from here:
    # http://stackoverflow.com/a/19156679/470844
    fixws = !"\
      if (! git diff-files --quiet .) && \
         (! git diff-index --quiet --cached HEAD) ; then \
        git commit -m FIXWS_SAVE_INDEX && \
        git stash save FIXWS_SAVE_TREE && \
        git rebase --whitespace=fix HEAD~ && \
        git stash pop && \
        git reset --soft HEAD~ ; \
      elif (! git diff-index --quiet --cached HEAD) ; then \
        git commit -m FIXWS_SAVE_INDEX && \
        git rebase --whitespace=fix HEAD~ && \
        git reset --soft HEAD~ ; \
      fi"
    # The different cases are:
    # - dirty tree and dirty index
    # - dirty tree and clean index
    # - clean tree and dirty index
    #
    # We have to consider separate cases because the 'git rebase
    # --whitespace=fix' is not compatible with empty commits (adding
    # '--keep-empty' makes Git not fix the whitespace :P).
    fixws-global-tree-and-index = !"\
      if (! git diff-files --quiet .) && \
         (! git diff-index --quiet --cached HEAD) ; then \
        git commit -m FIXWS_SAVE_INDEX && \
        git add -u :/ && \
        git commit -m FIXWS_SAVE_TREE && \
        git rebase --whitespace=fix HEAD~2 && \
        git reset HEAD~ && \
        git reset --soft HEAD~ ; \
      elif (! git diff-files --quiet .) ; then \
        git add -u :/ && \
        git commit -m FIXWS_SAVE_TREE && \
        git rebase --whitespace=fix HEAD~ && \
        git reset HEAD~ ; \
      elif (! git diff-index --quiet --cached HEAD) ; then \
        git commit -m FIXWS_SAVE_INDEX && \
        git rebase --whitespace=fix HEAD~ && \
        git reset --soft HEAD~ ; \
      fi"

```

**Comment by kanikaa1234** [comment]
Thanks @rumpelsepp!! This is helpful :smile:

**Comment by kanikaa1234** [comment]
@anikamurarka

**Comment by anikamurarka** [comment]
I will make the required changes, and will revert soon.

**Pull request closed without merging**